devenv/cloud/eks-cdk/utils/default-constants.ts

```diff
@@ -0,0 +1,34 @@
+
+
+
+export const DefaultConstants = {
+    AWS_ACCOUNT_ID: process.env.CDK_DEFAULT_ACCOUNT,
+    AWS_REGION: process.env.CDK_DEFAULT_REGION,
+    
+    CLUSTER_NAME: "sbtc-cluster",
+
+    INSTANCE_TYPE: "m5.large",
```

Comment:
How much thought has gone into the instance type? I don't think we have high resource requirements, but the blockchain nodes will probably require a lot of storage so we need a solution to support quite large volumes in the cluster.

Reply:
* `m5.large` is good enough for the resource requirements of the workloads we have. Out BTC Regtest node isn't going to consume a lot of compute/memory resources and all of our k8s `resource` limits are set to:

```
resources:
  requests:
    memory: 128Mi
    cpu: 50m
  limits:
    memory: 256Mi
    cpu: 100m
```

We have 11 deployments so here are the min and max bounds for the resources:

`min memory:  ~ 1.4 Gb`
`min cpu:  ~ 550 millicores`

`max memory:  ~ 2.75 Gb`
`max cpu:  ~ 1100 millicores`


I used this site to be able to calculate the total limit of pods given our resource requirements: 

https://learnk8s.io/kubernetes-instance-calculator

With our specs, we can run about 34 pods in one single node (realistically I expect around 29 pods)

<img width="1152" alt="m5 large-pods" src="https://github.com/stacks-network/sbtc/assets/35998898/2996584d-178b-4acd-aacb-970a6c4e7703">

Since this setup has roughly around 9 adding which corresponds to ~ 11 pods. So we can still comfortably run the rest of the workloads in 1 node, however, I've set the default to 2 nodes and the max to 4 nodes. This can easily be changed in the  `.env` file during deployment. 



* Regarding storage, I've installed the EBS Driver by default which can handle up to `64Tb` of storage allocation.

Reply:
If for any reason, the pod needs to consume extra resources, we can always add a VPA to them so that they can elastically respond to their resource needs.